src/lib/search-params.ts

```diff
@@ -29,49 +29,49 @@ const sortMapping = {
   },
 };
 
+const searchParamArraySchema = z
```

Comment:
You could consider using a name that's independent of the data type (in this case: 'array') and, instead, focuses on its function? The data type then is an implementation detail; if we would have to change the  type at some point, we don't have to rename the variable. For example: `searchParamFilterSchema`?